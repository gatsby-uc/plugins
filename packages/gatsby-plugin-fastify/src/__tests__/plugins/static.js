describe(`Gatsby Static Routes`, () => {
  it(`Should serve custom 404`, async () => {
    const response = await fastify.inject({
      url: "/badRoute",
      method: "GET",
    });

    expect(response.statusCode).toEqual(404);
    expect(response.payload).toMatchSnapshot();
  });

  it(`Should serve static index route`, async () => {
    const response = await fastify.inject({
      url: "/",
      method: "GET",
    });

    expect(response.statusCode).toEqual(200);
    expect(response.payload).toMatchSnapshot();
  });

  it(`Should serve static route with or without trailing /`, async () => {
    const noSlashResponse = await fastify.inject({
      url: "/posts/page-1",
      method: "GET",
    });

    const slashResponse = await fastify.inject({
      url: "/posts/page-1/",
      method: "GET",
    });

    expect(noSlashResponse.statusCode).toEqual(200);
    expect(slashResponse.statusCode).toEqual(200);
    expect(noSlashResponse.payload).toEqual(slashResponse.payload);
  });
});
