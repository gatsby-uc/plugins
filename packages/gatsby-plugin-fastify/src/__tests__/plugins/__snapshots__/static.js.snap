// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gatsby Static Routes Should serve custom 404 1`] = `
"<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Gatsby Plugin Fastify</title>
</head>

<body>
  <h1>Page Not Found</h1>
</body>

</html>"
`;

exports[`Gatsby Static Routes Should serve static index route 1`] = `
"<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Gatsby Plugin Fastify</title>
</head>

<body>
  <h1>Hello World</h1>
</body>

</html>"
`;
